File: 03_recursion/c++/02_greet.cpp
#include <iostream>

using namespace std;

void greet2(string name){
	cout << "how are you, " << name << "?" << endl;
}

void bye(){
	cout << "ok bye!" << endl;
}

void greet(string name){
	cout << "hello, " << name << "!" << endl;
	greet2(name);
	cout << "getting rady to say bye..." << endl;
	bye();
}

int main(int argc, char *argv[]) {
	greet("adit");
}

File: 03_recursion/java/02_greet.java
import java.util.*;
import java.lang.*;
import java.io.*;

class grokkingAlgorithms
{	
	public static void greet2(String name){
		System.out.println("how are you, " + name + "?");
	}
	
	public static void bye(){
		System.out.println("ok bye!");
	}
	
	public static void greet(String name){
		System.out.println("hello, " + name + "!");
		greet2(name);
		System.out.println("getting ready to say bye...");
		bye();		
	}
	
	
	public static void main (String[] args) throws java.lang.Exception
	{
		greet("adit");
	}
}